narrative_llm_agent/agents/workspace.py

```diff
@@ -48,120 +49,65 @@ def __init__(
             input must be a string with format number/number/number. Do not input a dictionary or a
             JSON-formatted string. This might take a moment to run, as it fetches data from a database."""
-            return self._get_report(process_tool_input(upa, "upa"))
+            ws = Workspace(token=self._token)
+            blobstore = Blobstore(token=self._token)
+            return get_report(process_tool_input(upa, "upa"), ws, blobstore)
 
         @tool("get-report-from-job-id", args_schema=JobInput, return_direct=False)
-        def get_report_from_job_id(job_id: str) -> str:
+        def get_report_from_job_id_tool(job_id: str) -> str:
             """Fetch a report object from a KBase Narrative."""
-            return self._get_report_from_job_id(process_tool_input(job_id, "job_id"))
+            ws = Workspace(token=self._token)
+            ee = ExecutionEngine(token=self._token)
+            blobstore = Blobstore(token=self._token)
+            return get_report_from_job_id(
+                process_tool_input(job_id, "job_id"), ee, ws, blobstore
+            )
 
         @tool("get-object", args_schema=UpaInput, return_direct=False)
-        def get_object(upa: str) -> str:
+        def get_object_tool(upa: str) -> dict:
             """Fetch a particular object from a KBase Narrative. This returns a JSON-formatted data object
             from the Workspace service. Its format is dependent on the data type. The upa input must be a
             string with format number/number/number. Do not input a dictionary or a JSON-formatted string."""
-            return self._get_object(process_tool_input(upa, "upa"))
+            ws = Workspace(token=self._token)
+            return ws.get_objects([process_tool_input(upa, "upa")])[0]
 
         @tool("Get object name", return_direct=False)
-        def get_object_name(upa: str) -> str:
+        def get_object_name_tool(upa: str) -> str:
             """Get the name of a data object from its UPA. This returns the name string. An UPA input must
             be a string with the format number/number/number."""
-            return self._get_object_name(upa)
+            ws = Workspace(token=self._token)
+            return ws.get_object_info(process_tool_input(upa, "upa"))["name"]
 
         self.agent = Agent(
             role=self.role,
             goal=self.goal,
             backstory=self.backstory,
             verbose=True,
             tools=[
-                list_objects,
-                get_object,
-                get_report,
-                get_report_from_job_id,
-                get_object_name,
+                list_objects_tool,
+                get_object_tool,
+                get_report_tool,
+                get_report_from_job_id_tool,
+                get_object_name_tool,
             ],
             llm=self._llm,
             allow_delegation=False,
             memory=True,
         )
-
```

Comment:
These are all removed.
* _list_objects
* _get_object
* _get_object_name
Are effectively one-liners and just migrated to the tool wrapper.
`get_report` and `get_report_from_job_id` (along with everything else in the WorkspaceUtil class) are moved to the new report_tools module.